frontend/src/store/actions/user/user.js

```diff
@@ -0,0 +1,114 @@
+import axios from "axios";
+import { userConstants } from "../actionTypes";
+
+// get a single user
+const getUserByUserIdSuccess = (user) => ({
+    type: userConstants.GET_USER,
+    target: user,
+});
+
+const getUserByUserIdFailure = (error) => {
+    const actionType = error.response.status === 404
+        ? userConstants.GET_USER_FAILURE_USER_NOT_EXIST : null;
+    return {
+        type: actionType,
+        target: error,
+    };
+};
+
+export const getUserByUserId = (userId) => (dispatch) => axios.get("api/user", { params: { id: userId } })
```

Comment:
Other actions(auth, collection, ..) get arguments in the form of { id: user_id } from containers or components (so, userId should be {id: user_id}) and deliver parameters to axios like { params: userId }. Also, it would be better to add "/" to the "api/user".

Reply:
e.g.) export const getCollection = (collectionId) => (dispatch) => axios.get("/api/collection", { params: collectionId })
    .then((res) => { dispatch(getCollectionSuccess(res.data)); })
    .catch((err) => { dispatch(getCollectionFailure(err)); });

Reply:
I agreed. Using `api/user` without leading '/' is dangerous because it will use relative url.